exegol/model/ExegolContainer.py

```diff
@@ -334,16 +338,59 @@ def __applyXhostACL(self):
                 logger.error(f"The [green]xhost[/green] command is not available on your [bold]host[/bold]. "
                              f"Exegol was unable to allow your container to access your graphical environment ({debug_msg}).")
                 return
+            
+            logger.debug(f"DISPLAY variable: {GuiUtils.getDisplayEnv()}")
 
-            if EnvInfo.isMacHost():
-                logger.debug(f"Adding xhost ACL to localhost")
-                # add xquartz inet ACL
-                with console.status(f"Starting XQuartz...", spinner_style="blue"):
-                    os.system(f"xhost + localhost > /dev/null")
+            display_host = GuiUtils.getDisplayEnv().split(':')[0]
+            if display_host=='':
+                logger.debug("Connecting to container from local GUI, no X11 forwarding to set up")
+                # TODO verify that the display format is the same on macOS, otherwise might not set up xauth and xhost correctly 
+                if EnvInfo.isMacHost():
+                    logger.debug(f"Adding xhost ACL to localhost")
+                    # add xquartz inet ACL
+                    with console.status(f"Starting XQuartz...", spinner_style="blue"):
+                        os.system(f"xhost + localhost > /dev/null")
+                else:
+                    logger.debug(f"Adding xhost ACL to local:{self.config.getUsername()}")
+                    # add linux local ACL
+                    os.system(f"xhost +local:{self.config.getUsername()} > /dev/null")
+                return
+            
+            if shutil.which("xauth") is None:
+                if EnvInfo.is_linux_shell:
+                    debug_msg = "Try to install the package [green]xorg-xauth[/green] or maybe you don't have X11 on your host?"
+                else:
+                    debug_msg = "or you don't have one"
+                logger.error(f"The [green]xauth[/green] command is not available on your [bold]host[/bold]. "
+                             f"Exegol was unable to allow your container to access your graphical environment ({debug_msg}).")
+                return
+            
+            if display_host=="localhost" and self.config.getTextNetworkMode() != "host":
+                logger.warning("X11 forwarding won't work on a bridged container unless you specify \"X11UseLocalhost no\" in your host sshd_config")
+                logger.warning("[red]Be aware[/red] changing \"X11UseLocalhost\" value can [red]expose your device[/red], correct firewalling is [red]required[/red]")
+                logger.warning("The following documentation can be usefull to limit the exposure of your x11 socket: https://studioware.com/wikislax/index.php?title=X11_over_the_network#X11_firewalling")
+                return
+
+            # Setting up the temporary file to pass the xauth cookie to the container
+            tmpXauthority = f"/tmp/.tmpXauthority{random.randint(0, 999)}"
```

Comment:
As is, the code is specific for linux platform `/tmp` path hardcoded wont work on Windows for exemple.
If its intended and linux specific we can add a restriction, overwise we will have to use tempfile (https://docs.python.org/3/library/tempfile.html) and see if `xauth extract` can work with an existing file or customizing some options.
These changes are not mandatory for this PR, i'll be able to add this at a later date. (but you can add a `#TODO` here for me to remember.

Reply:
This is a very good point. Because the previous function excluded (and still excludes) Windows environments I took the liberty of hardcoding `/tmp`. But you are right to rely on tested libraries to make more generic code so I changed the implementation.